api/app/models/env_canada.py

```diff
@@ -169,48 +174,105 @@ def flag_file_as_processed(session, url):
     session.commit()
 
 
+def check_if_model_run_complete(
+        session: Session,
+        model: ModelEnum,
+        projection: str,
+        now: datetime.datetime,
+        hour: int):
+    """ Check if a particular model run is complete """
+
+    # Get all the download urls for a particular model run.
+    urls = get_model_run_download_urls(now, hour)
+    # Start off assuming it's complete, then attempt to falsify assumption.
+    complete = True
+    # Iterate through url's.
+    for url, _ in urls:
+        # Check if url as been processed.
+        processed_file_record = get_processed_file_record(session, url)
+        # If the url has not been processed, mark as false and stop.
+        if not processed_file_record:
+            logger.info(
+                'for model %s:%s run %s, url %s not yet processed', model, projection, hour, url)
+            complete = False
+            break
```

Comment:
`SELECT url FROM processed_model_run_urls WHERE url IN ( 'url1', 'url2', ... );` would be more suitable but I would say this is nice to have

Reply:
It would be a ```SELECT count(url)``` and then check if the number is the same. Agreed it would be faster - I thought about it but didn't like a) writing more code and b) sending such a monster query down - going to be a really big query...

Reply:
Re-factored - I like it more now.